moquette-0.16/broker/src/main/java/io/moquette/broker/PostOffice.java

```diff
@@ -352,6 +354,7 @@ RoutingResults receivedPublishQos1(MQTTConnection connection, Topic topic, Strin
         final String clientId = connection.getClientId();
         if (!authorizator.canWrite(topic, username, clientId)) {
             LOG.error("MQTT client: {} is not authorized to publish on topic: {}", clientId, topic);
+            connection.dropConnection();
```

Comment:
do we want this after `ReferenceCountUtil.release(msg);` similar to above? not sure the implications

Reply:
Doesn't matter, that's just for reference counting, as long as it hits 0 eventually we're good